### ISSUE TITLE
Doesn't work "ignore-leading" well with "implementation-comments=join"

### ISSUE THREAD

**Issue by kfilippenok** [pasdoc/pasdoc#210]
Tested on latest _master_ branch and _snapshot_ versions.

Config file:
```
title=My project
language=ru.utf8
write-uses-list
auto-back-comments
auto-abstract
implementation-comments=join
ignore-leading=-
use-tipue-search
html-head=head.html
introduction=introduction.txt
output=site/
css=styles.css
```

Interface section:
```
// Test function.
function testfnc(arg1: Integer; arg2: String): Boolean; 
```
Implementation section:
```
//--------------------------------------------------------------------------------
// Test function.
//
// @param arg1 Argument 1
// @param arg2 Argument 2
// @return @True or @False
//--------------------------------------------------------------------------------
function testfnc(arg1: Integer; arg2: String): Boolean;
begin
  result := False;
end; 
```

Generated docs:
![image](https://github.com/user-attachments/assets/250e3185-89a8-4eae-9832-1e728535f2ff)

**Comment by michaliskambi** [comment]
Thanks for the report. I see the problem is not really related to implementation-comments=join, it seems we don't apply `ignore-leading` to single-line comments `//` at all.

When using multi=line `{ ... }` it's all OK:

```delphi
unit ok_ignore_leading_impl;

interface

{ --------------------------------------------------------------------------------
  Test function.

  @param arg1 Argument 1
  @param arg2 Argument 2
  @return @True or @False
 -------------------------------------------------------------------------------- }
function testfnc_only_intf(arg1: Integer; arg2: String): Boolean;

// Test function.
function testfnc(arg1: Integer; arg2: String): Boolean;

implementation

{ --------------------------------------------------------------------------------
  Test function.

  @param arg1 Argument 1
  @param arg2 Argument 2
  @return @True or @False
  -------------------------------------------------------------------------------- }
function testfnc(arg1: Integer; arg2: String): Boolean;
begin
  result := False;
end;

end.
```

Doing 

```
pasdoc ok_ignore_leading_impl.pas --output=/tmp/ --ignore-leading=- 
```

shows everything is OK.

In contrast, using `//` comments fails, also for things only in the interface:

```delphi
unit ok_ignore_leading_impl;

interface

//--------------------------------------------------------------------------------
// Test function.
//
// @param arg1 Argument 1
// @param arg2 Argument 2
// @return @True or @False
//--------------------------------------------------------------------------------
function testfnc_only_intf(arg1: Integer; arg2: String): Boolean;

// Test function.
function testfnc(arg1: Integer; arg2: String): Boolean;

implementation

//--------------------------------------------------------------------------------
// Test function.
//
// @param arg1 Argument 1
// @param arg2 Argument 2
// @return @True or @False
//--------------------------------------------------------------------------------
function testfnc(arg1: Integer; arg2: String): Boolean;
begin
  result := False;
end;

end.
```

The result is wrong with

```
pasdoc ok_ignore_leading_impl.pas --output=/tmp/ --implementation-comments=join --ignore-leading=- 
```

and it seems the `implementation-comments=...` value here doesn't matter, it's always wrong.

Looking into it....

**Comment by michaliskambi** [comment]
Fixed!

I saw this was deliberate in the code (`--ignore-leading` was affecting only multi-line comments) but without any explanation "why", and I didn't find an explanation either in the commit history.

So I just enabled `--ignore-leading` for all comments, added auto-test ( https://github.com/pasdoc/pasdoc/blob/master/tests/testcases/ok_ignore_leading_impl.pas ), and made sure it doesn't break any existing test.

The new version is already available in snapshots ( https://github.com/pasdoc/pasdoc/releases/tag/snapshot ).
